spec/compiler/type_inference/union_spec.cr

```diff
@@ -58,4 +58,62 @@ describe "Type inference: union" do
       LibC::Foo.new.a
       ), flags: "some_flag") { int32 }
   end
+
+  it "types union" do
+    assert_type(%(
+      Union(Int32, String)
+      )) { union_of(int32, string).metaclass }
+  end
+
+  it "types union of same type" do
+    assert_type(%(
+      Union(Int32, Int32, Int32)
+      )) { int32.metaclass }
+  end
+
+  it "can reopen Union" do
+    assert_type(%(
+      struct Union
+        def self.foo
+          1
+        end
+      end
+      Union(Int32, String).foo
+      )) { int32 }
+  end
+
+  it "can reopen Union and access T" do
+    assert_type(%(
+      struct Union
+        def self.types
```

Comment:
is types available in the std? it's a bit confusing that the same method is used in the next spec.